core/src/main/java/org/apache/iceberg/avro/ValueWriter.java

```diff
@@ -27,7 +27,5 @@
 public interface ValueWriter<D> {
   void write(D datum, Encoder encoder) throws IOException;
 
-  default Stream<FieldMetrics> metrics() {
-    return Stream.empty(); // TODO will populate in following PRs
-  }
+  Stream<FieldMetrics> metrics();
```

Comment:
`FieldMetrics` is parameterized, but this is a bare reference. Could you update it? I think it should be `FieldMetrics<?>` since the metrics are not necessarily for the written value type, `D`.